rmw_zenoh_cpp/src/detail/liveliness_utils.hpp

```diff
@@ -237,8 +237,12 @@ std::string zid_to_str(const z_id_t & id);
 }  // namespace liveliness
 
 ///=============================================================================
-// Helper function to generate a randon GID.
+/// Helper function to generate a randon GID.
 void generate_random_gid(uint8_t gid[RMW_GID_STORAGE_SIZE]);
+
+///=============================================================================
+/// Generate a hash for a given GID.
+size_t hash_gid(const uint8_t * gid);
```

Comment:
I know you are just copying this over, but I'm going to suggest that we change this signature to:

```suggestion
size_t hash_gid(const uint8_t gid[RMW_GID_STORAGE_SIZE]);
```

It won't make much of a practical difference, but it will be clear to the reader exactly what the size of gid is.